src/Microsoft.Owin.Security.Authorization/IAuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public interface IAuthorizationDependenciesProvider
+    {
+        Func<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnCreate { get; }
```

Comment:
Maybe I'm missing something - where does the creation of an extra class come in?  It's entirely legal to declare a method `public T3 Foo(T1 arg1, T2 arg2) { }` and pass it as an argument directly to another method `Bar(Func<T1, T2, T3> foo) { }`...